version/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+# Generate a O2Version header and library that contain a couple of useful
+# fields such as the git commit with which we build.
+# We can also use this to define VERSION MACROS etc.
+
+include(O2NameTarget)
+
+execute_process(COMMAND git log --oneline -1 --pretty=format:%h
```

Comment:
That's a minor point, but to be more flexible and support installation from a non-git source (e.g. from a tarball) you might want to use the RESULT_VARIABLE option to check if the git command was actually successful

Reply:
Thanks. Done.